histomicstk/segmentation/membranes/membrane_detection.py

```diff
@@ -0,0 +1,119 @@
+import histomicstk.filters.shape as htk_shape_filters
+
+import numpy as np
+import scipy as sp
+
+import skimage.color
+import skimage.filters
+import skimage.io
+import skimage.measure
+import skimage.morphology
+
+
+def membrane_detection(I, min_sigma=1, max_sigma=3, beta=4, c=2,
+                       f_threshold=0.4, min_lsize=100,
+                       b_dilation=1, b_split_dilation=3):
+    """Performs membrane detection.
+    Takes as input a dab-deconvolved image and performs the membrane filtering
+    using eigenvalue analysis of scale-space blur image Hessians to detect
+    membranes. This also detects membrane branches using bifurcation patterns.
+
+    Parameters
+    ----------
+    I : array_like
+        A DAB intensity image obtained from ColorDeconvolution.
+    min_sigma : int
+        Minimum vector of gaussian deviations used for membrane filter.
+        Default value = 1.
+    max_sigma : int
+        Maximum vector of gaussian deviations used for membrane filter.
+        Default value = 3.
+    beta : double
+        Scaling property of deviation from membrane. Decrease to increase
+        importance of deviation in final score. Default value = 4.
+    c : double
+        Scaling property of quadratic curvature magnitude. Decrease to
+        increase the importance of curvature magnitude in final score.
+        Default value = 2.
+    f_threshold : double
+        A foreground threshold of membrane. Used with threshold_otsu.
+        Default value = 0.4.
+    min_lsize : int
+        Minimum label size to be removed. Default value = 100.
+    min_lsize : int
+        Minimum label size to be removed. Default value = 100.
+    b_dilation : int
+        Branch dilation factor. Default value = 1.
+    b_split_dilation : int
+        Branch dilation factor to be splited. Default value = 3.
+
+    Returns
+    -------
+    Label : array_like
+        Label image where positive values correspond to membrane labels.
+    Split : array_like
+        Splitted image where positive values correspond to splitted points.
+    Branches : array_like
+        Branche image of where positive values correspond to membrane branches.
+    """
+
+    # membrane filtering
+    im_deviation, im_thetas = htk_shape_filters.membranefilter(
+        I, np.arange(min_sigma, max_sigma+1),
+        beta, c
+    )
```

Comment:
It uses the same [logic](https://github.com/scikit-image/scikit-image/blob/master/skimage/filters/_frangi.py#L74).